### PR TITLE
fix xcb features

### LINKED ISSUE #130: xcb features:  method `from_glib_none` is not a member of trait `FromGlibPtrFull`
According to the [doc](http://gtk-rs.org/docs/glib/translate/trait.FromGlibPtrFull.html?search=FromGlibPtr), we have now multiples traits for each method. Here is the error after putting this in my ```Cargo.toml```:
```
[dependencies.cairo-rs]
version = "0.1.3"
default-features = false
features = ["xcb"]
```
```
error[E0407]: method `from_glib_none` is not a member of trait `FromGlibPtrFull`
  --> /home/charles/.cargo/registry/src/github.com-1ecc6299db9ec823/cairo-rs-0.1.3/src/xcb.rs:42:5
   |
42 |       unsafe fn from_glib_none(ptr: *mut ffi::xcb_connection_t) -> XCBConnection {
   |  _____^ starting here...
43 | |         assert!(!ptr.is_null());
44 | |         XCBConnection(ptr)
45 | |     }
   | |_____^ ...ending here: not a member of trait `FromGlibPtrFull`

error[E0407]: method `from_glib_none` is not a member of trait `FromGlibPtrFull`
  --> /home/charles/.cargo/registry/src/github.com-1ecc6299db9ec823/cairo-rs-0.1.3/src/xcb.rs:80:5
   |
80 |       unsafe fn from_glib_none(ptr: *mut ffi::xcb_render_pictforminfo_t) -> XCBRenderPictFormInfo {
   |  _____^ starting here...
81 | |         assert!(!ptr.is_null());
82 | |         XCBRenderPictFormInfo(ptr)
83 | |     }
   | |_____^ ...ending here: not a member of trait `FromGlibPtrFull`

error[E0277]: the trait bound `xcb::XCBConnection: glib::translate::FromGlibPtrNone<*mut ffi::xcb_connection_t>` is not satisfied
  --> /home/charles/.cargo/registry/src/github.com-1ecc6299db9ec823/cairo-rs-0.1.3/src/xcb.rs:62:18
   |
62 |         unsafe { from_glib_none(self.to_glib_none().0) }
   |                  ^^^^^^^^^^^^^^ the trait `glib::translate::FromGlibPtrNone<*mut ffi::xcb_connection_t>` is not implemented for `xcb::XCBConnection`
   |
   = note: required by `glib::translate::from_glib_none`

error[E0277]: the trait bound `xcb::XCBRenderPictFormInfo: glib::translate::FromGlibPtrNone<*mut ffi::xcb_render_pictforminfo_t>` is not satisfied
   --> /home/charles/.cargo/registry/src/github.com-1ecc6299db9ec823/cairo-rs-0.1.3/src/xcb.rs:100:18
    |
100 |         unsafe { from_glib_none(self.to_glib_none().0) }
    |                  ^^^^^^^^^^^^^^ the trait `glib::translate::FromGlibPtrNone<*mut ffi::xcb_render_pictforminfo_t>` is not implemented for `xcb::XCBRenderPictFormInfo`
    |
    = note: required by `glib::translate::from_glib_none`

error: aborting due to 2 previous errors

error: Could not compile `cairo-rs`.
Build failed, waiting for other jobs to finish...
error: build failed
```

### PR DESCRIPTION
**Pull request by charlesvdv**
fix #130

### CODE REVIEW

**Comment by EPashkin** [comment]
Thanks for fix, but as currently cairo can be used without glib (see https://github.com/gtk-rs/cairo/pull/127) this module need more fixes

**Comment by charlesvdv** [comment]
Alright ! If I implements the fix from #127, you could merge this PR because I really need to use the cairo-xcb feature ?

**Comment by EPashkin** [comment]
Sure, It was your mistake that we forgot change this feature.
And I don't know `xcb` to check if it work normally.

File: src/xcb.rs [hunk 1]
```diff
@@ -59,33 +79,52 @@ impl AsRef<XCBConnection> for XCBConnection {
 
 impl Clone for XCBConnection {
     fn clone(&self) -> XCBConnection {
-        unsafe { from_glib_none(self.to_glib_none().0) }
+        unsafe { Self::from_raw_none(self.to_raw_none()) }
```
Comment (charlesvdv) [inline src/xcb.rs]: I don't really know if it's better to remove the ```unsafe``` here or leave it there and mark the impl ```unsafe``` like #127 ?

Comment (EPashkin) [inline src/xcb.rs]: all new from_raw* functions need `unsafe` as in other files.

Comment (charlesvdv) [inline src/xcb.rs]: alright!

**Comment by EPashkin** [comment]
You using cairo without glib?

**Comment by charlesvdv** [comment]
@EPashkin I'm not very sure yet :/ I'm trying to get this PR merged before playing with cairo (and xcb) because the library don't even compile without 6824b24.

**Comment by EPashkin** [comment]
You can use http://doc.crates.io/specifying-dependencies.html#overriding-dependencies to point to your fork of cairo and test

**Comment by charlesvdv** [comment]
Yes I know but what if others peoples wants to use the xcb features? You want me to test it before merging with a dummy app using the cairo-xcb API ?

**Comment by GuillaumeGomez** [comment]
It's as simple as follows: we won't merge until we're sure it works.

**Comment by EPashkin** [comment]
Code looks good for me, only last is squash. And I can't even check that it builds due no xcb on Windows.

**Comment by charlesvdv** [comment]
I've been trying to get cairo with xcb working but no luck so far. [My test](https://gist.github.com/charlesvdv/de36473893913f7ebbf342047ea8c4ae) has been tried with and without ```glib``` and even without my last commit and I still get an invisible window. The test is just a transcription of [xcbsurface.c](https://www.cairographics.org/cookbook/xcbsurface.c/) which works great on my computer. 

Is it possible that the xcb backend never worked  (looking at the #102 comments, it was never tested)? As I don't totally understand how ffi works in Rust, I don't know if it's possible or not.

NOTE: I don't want to criticize the work you are doing because I found it awesome 👍

**Comment by GuillaumeGomez** [comment]
That's the big lacking part of gtk-rs crates: not enough tests. Testing is quite difficult, and when we encounter an issue, we add the corresponding test (and even then, it's really difficult). If you ever happen to find the issue, please fix it and add a test. :)

Otherwise sorry but I'm quite busy so I won't be able to help in the days/weeks to come.

**Comment by charlesvdv** [comment]
No problem I totally understand ! 😄 I will try to look into it. Otherwise I will try to use the xlib backend.

**Comment by GuillaumeGomez** [comment]
Any advances on this issue would be great. Please keep us up to date on your tries. 😃

**Comment by mjkillough** [comment]
You should pass `win` to `cairo::XCBSurface()`, not [`screen.ptr`](https://gist.github.com/charlesvdv/de36473893913f7ebbf342047ea8c4ae#file-main-rs-L82). Once I made that change, your test drew correctly both with and without the `glib` feature.

This pull request is exactly what I need - thank you!

**Comment by charlesvdv** [comment]
@mjkillough Thanks to you! I'm going to update the gist and I confirm it's working !

**Comment by charlesvdv** [comment]
@GuillaumeGomez since it's working, what do you think of the PR ?

**Comment by GuillaumeGomez** [comment]
For me it's good!

@EPashkin: Did you check this one? :)

**Comment by charlesvdv** [comment]
I think he did at  https://github.com/gtk-rs/cairo/pull/131#issuecomment-301124090 :)

**Comment by GuillaumeGomez** [comment]
Indeed! Thanks!

**Pull request merged**